### DISCUSSION TITLE
[LGTM] Failing Python Builds

### DISCUSSION THREAD

**Discussion by sasobadovinac** [github/securitylab#206] [Q&A] [answered]
Hi,

we have set up the LGTM builds for the FreeCAD project and have been successfully running it now for some time, lately however all the Python builds are failing because of timing out, before the builds have finished in about an hour... Can we please ask for some help in better understanding why this is a problem now and how to fix it? 

https://lgtm.com/projects/g/FreeCAD/FreeCAD/?mode=list

Thanks and have a nice day!

**Comment by RasmusWL** [comment, marked as answer]
Hi @sasobadovinac, thank you for your report :+1: we've introduced some new CodeQL libraries recently in the Python analysis, and know about some projects that this have caused problems for. We're working on getting this resolved, but sadly nothing you can do on your part.

**Comment by tausbn** [comment]
Hey, just as a brief update, LGTM analysis for FreeCAD should now be up and running again.

Our apologies for accidentally breaking the analysis for you. We usually try _not_ to break things for people who are actively using our services, but in this case the FreeCAD project simply slipped under our radar. (Rest assured that going forward we'll be keeping an eye on it!)

**Reply by sasobadovinac** [reply]
Great! We are looking forward to continue using it. As a side note I can also say that we have started to successfully make manual builds and analysis of our C++ code. Thanks again for the great tools and services!

**Reply by tausbn** [reply]
@sasobadovinac I'm terribly sorry for the inconvenience, but it seems we ended up breaking the analysis for FreeCAD again. :disappointed: 
It seems we were not as vigilant with regard to FreeCAD as I had promised...

I expect the analysis to start working again in a couple of weeks time when we do the next upgrade on LGTM.com. In the meantime, you may be able to get the analysis to start working again by temporarily disabling all of the security queries in your `lgtm.yml` config file (let me know if you would like assistance with this).

Also, I'm adding FreeCAD to our standard regression tests, so we hopefully avoid this happening again in the future.
